Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +12,198 @@ public static class BalanceStats
+            {
+                for (int i = 0; i < personBalances.Value.Count - 1; i++)
+                {
+                    var balanceDifferences = personBalances.Value[i + 1] - personBalances.Value[i];
+                    if (balanceDifferences < biggestLostEver)
+                    {
+                        names.Clear();
+                        biggestLostEver = balanceDifferences;
+                        names.Add(personBalances.Key);
+                        continue;
+                    }
+                    if (balanceDifferences == biggestLostEver)
+                    {
+                        names.Add(personBalances.Key);
+                    }
+                }
+            }
+
+            if (names.Count == 0)
+            {
+                return "N/A.";
+            }
+            return $"{BuildNamesString(names)} lost the most money. {BuildCurrancy(biggestLostEver)}.";
+        }
         /// <summary>
         /// Return name and current money of the richest person.
         /// </summary>
         public static string FindRichestPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (peopleAndBalances == null || peopleAndBalances.Length == 0)
+                return "N/A.";
+
+            Dictionary<string, List<float>> peopleBalances = ArrayOfPeople(peopleAndBalances);
+            List<string> names = new List<string>();
+            var biggestLastBalance = float.MinValue;
+
+            foreach (var personBalances in peopleBalances)
+            {
+                {
+                    if (personBalances.Value.Last() > biggestLastBalance)
+                    {
+                        names.Clear();
+                        biggestLastBalance = personBalances.Value.Last();
+                        names.Add(personBalances.Key);
+                    }
+                    else if (personBalances.Value.Last() == biggestLastBalance)
+                    {
+                        names.Add(personBalances.Key);
+                    }
+                }
+            }
+
+            string peopleString = BuildNamesString(names);
+            if (names.Count > 1)
+            {
+                return $"{peopleString} are the richest people. {BuildCurrancy(biggestLastBalance)}.";
+            }
+            return $"{peopleString} is the richest person. {BuildCurrancy(biggestLastBalance)}.";
         }
+       
 
         /// <summary>
         /// Return name and current money of the most poor person.
         /// </summary>
         public static string FindMostPoorPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (peopleAndBalances == null || peopleAndBalances.Length == 0)
+                return "N/A.";
+
+            Dictionary<string, List<float>> peopleBalances = ArrayOfPeople(peopleAndBalances);
+            List<string> names = new List<string>();
+            StringBuilder peopleString = new StringBuilder();
```

Comment:
Codacy found an issue: [Remove the unused local variable 'peopleString'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5838869)